order/views.py

```diff
@@ -103,6 +103,10 @@ def orders(self, request):
 
         try:
             orders = order_manager.fetch_orders(order_id=validated_data.get("order_id"))
+
+            if orders is None:
+                raise RuntimeError("Order id not found")
```

Comment:
raising RuntimeError would be incorrect here.
raise Exception.

Reply:
ohh hold on why are we raising an exception in viewset? 
Where will it be caught ?

Reply:
raise the Exception in fetch_orders so that it will be caught in **def orders()** viewset

Reply:
it will caught in orders() it self because  we are using try catch block

Reply:
okay but this is still not the right way to do it. empty checks should be done in the internal functions itself.
Or you can instead not raise, just return something `if orders is None:`